pkg/proxy/iptables/proxier.go

```diff
@@ -280,7 +292,7 @@ func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
 		// Hunt for service and endpoint chains.
 		for chain := range existingNATChains {
 			chainString := string(chain)
-			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") {
+			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") || strings.HasPrefix(chainString, "KUBE-XLB-") {
```

Comment:
Girish, we need a TODO for nodeports, too, right?

On Tue, Aug 16, 2016 at 5:47 PM, Tim Hockin <thockin@google.com> wrote:

> I figured KUBE-LB and KUBE-FW would come together to be "all rules
> processed only for external VIPs"
>
> On Tue, Aug 16, 2016 at 3:49 PM, Minhan Xia <notifications@github.com>
> wrote:
>
>> In pkg/proxy/iptables/proxier.go
>> <https://github.com/kubernetes/kubernetes/pull/29409#discussion_r75035276>
>> :
>>
>> > @@ -280,7 +292,7 @@ func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
>> >  		// Hunt for service and endpoint chains.
>> >  		for chain := range existingNATChains {
>> >  			chainString := string(chain)
>> > -			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") {
>> > +			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") || strings.HasPrefix(chainString, "KUBE-XLB-") {
>>
>> I think this PR can move the MARK-MASQ logic from KUBE-FW to KUBE-LB.
>>
>> Then KUBE-FW is truly a firewall chain.
>>
>> —
>> You are receiving this because you were assigned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/29409/files/b66ec449901a52102d485fe6ff1b7869282bc55c#r75035276>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AFVgVAlSK7raAY9jYYg2Av0oszyo4pEuks5qgj5pgaJpZM4JSUXH>
>> .
>>
>
>